spock-specs/src/test/groovy/spock/util/JvmSpec.groovy

```diff
@@ -13,9 +13,11 @@
  */
 package spock.util
 
+import spock.lang.Isolated
 import spock.lang.Specification
 import spock.util.environment.*
 
+@Isolated
```

Comment:
`@RestoreSystemProperties` is usually applied in the specifications which globally change state. I wonder, how much it should be advised for those specification to use `@Isolated` or even have it enabled by default (with an option to override it)?

On the other hand. very often those changes might not impact the other part of the test suite (such as that one with `java.specification.version`), so it might be to restricting to isolate them by default. Maybe it (and similar built-in extensions) should be clearly added to the section describing which kind of tests (with which extensions) should be considered for manual isolation when parallel mode is enabled?

Btw, without `@Isolated` here there were test failures or you just added it for peace of mind?

Reply:
>  Maybe it (and similar built-in extensions) should be clearly added to the section describing which kind of tests (with which extensions) should be considered for manual isolation when parallel mode is enabled?

Could you expand on what you mean here exactly. Have you read the documentation that I added and are missing something or is this comment without having looked at the docs yet? 

As documented `@RestoreSystemProperties` will write lock on system properties, if you are changing some property that has global impact you can then add `@Isolated`. In this case it affected several other tests that depended on the java version.

Reply:
> Have you read the documentation that I added and are missing something or is this comment without having looked at the docs yet?

The diff for "parallel_execution.adoc" was folded and I completely missed it...

Reply:
I'd suggest to look at the rendered documentation